### ISSUE TITLE
GoDaddy Implementation: 404 error on cgiNodeInfo.jscript

### ISSUE THREAD

**Issue by JaredJames** [UeiRicho/cgi-node#13]
I followed the GoDaddy Shared Hosting Documentation.

download latest node file (no ext)
download optimized cgi-node.js

chmod 755 on upload through cPanel File Manager

In the docs it says to change the <account name> on line 1 to your GoDaddy account name. That line reflects an installation path on the D:\ drive and points to node.exe. Obviously, since the docs tell you to download the 64 bit linux distro it doesn't have an exe file it has node (no extension). There is also no <accountname> placeholder in the first line.

I tried to point to the node file by editing that line to reflect /home/<account name>/path/to/node with no luck. Can you tell me how to get node running as cgi application?

I have completed all other steps on the docs.

**Comment by deda-ca** [comment]
Assuming you have downloaded node.js into the following folder: /home/<username>/node.js/
The first line of cgi-node.js file should look exactly like this on Godaddy:

#!/home/<username>/node.js/node

Where <username> is replaced by your account username. It is important to start on the first line at the first character with #! followed by the full path to the node executable. Also note that there is no quotes around the path in linux (GoDaddy).

I should also point out that you'll need to update the SessionPath as well to point to a local folder that exists. For example: SessionPath: "/home/<username>/node.js/sessions/"

If you have downloaded the cgi-node.min.js you'll find the variable at the end of the third line containing the following value: SessionPath:"D:/Programs/nodejs/sessions/"

Session support has implemented in the latest release and the documentation needs to be updated to reflect the changes.

I hope this helps. If you have any questions please don't hesitate to ask. Also let me if you are still having any issues.

**Comment by JaredJames** [comment]
Thank you for your reply and support!

cgi-node.js is in /home/accountname/public_html/cgi-bin/
node is in /home/accountname/bin/nodeJS/

First line of cgi-node.js is #!"/home/accountname/bin/nodeJS/node"
Session path is SessionPath:"home/accountname/bin/nodeJS/sessions"
That folder does exist on the server in the nodeJS directory.

All files are 755, accountname is actually my GoDaddy Account name in cgi-node.js.
Added lines to .htaccess
Created cgiNodeInfo.jscript, point browser to mydomain.com/cgiNodeInfo.jscript and I am receiving a 404 apache error.

Do you know what might cause this?

**Comment by deda-ca** [comment]
Try removing the quotes from the first line of cgi-node.js. 
Before: #!"/home/accountname/bin/nodeJS/node"
After: #!/home/accountname/bin/nodeJS/node

I get an error if there are quotes around the path.

Also, if you are using the extension .jscript for the server side script files make sure that the .htaccess file reflects this. For example, your .htaccess file should look like this:

Action     cgi-node     /cgi-bin/cgi-node.js 
AddHandler cgi-node     .jscript

Finally you might want to also update CgiNodeConfig. EmbededScriptExtensions to include '.jscript'. For example: within cgi-node.js, find var CgiNodeConfig and add '.jscript' to EmbededScriptExtensions or replace '.jss' with '.jscript'

var CgiNodeConfig = 
{
    ...
    EmbededScriptExtensions: ['.jss', '.jscript'],
    ...
};

**Comment by JaredJames** [comment]
I have made these changes and still get a 500 error and 404. At a loss! I can provide cpanel login details if you would like / mind taking a look. Woth noting in the docs it said node would be 11MB it is currently 20+

**Comment by deda-ca** [comment]
Yes, you are right, the latest version of node.js is 20+ MB for 64 bit Linux. I don't mind having a look at it. We can also do google hangle out with screen sharing if you like.

**Comment by agryaznov** [comment]
Hello. I'm getting the 500 error too, with this message in server log: 

"AH01215: suexec policy violation: see suexec log for more details: /home/user/public_html/cgi-bin/cgi-node.min.js"

Unfortunately, I cannot access suexec log on the godaddy shared hosting to investigate what is the problem in details.

Any help on the issue would be appreciated. Thx.

**Comment by deda-ca** [comment]
We did get JaredJames setup working. Thank you JaredJames for the opportunity. 

Here is a list of things to look out for:
- The node.js execute file must match the system architecture. 64bit or 32bit. Make sure to download the correct one. On Godaddy servers it is the Linux 64bit version of node.js.
- The first line within cgi-node.min.js or cgi-node.js that points to the node.js must be the full exact path to the node executable with a leading '/' and no quotes around it (If running under windows then you need the double quotes, but on Godaddy Linux servers make sure there are no quotes). 

Here are some tips for the node path under Godaddy linux servers (Assuming you've placed node under this path:/home/accountname/bin/node)
- CORRECT format is:#!/home/accountname/bin/node
- Make sure there are no leading spaces or lines.
- #!home/accountname/bin/node : INCORRECT, missing starting '/' 
- #!"/home/accountname/bin/node" : INCORRECT, must NOT have quotes around the path.
- #!home/account/bin/node or #!~/bin/node or #~bin/node - INCORRECT, must be the full exact path starting with '/'
- The session path has to be updated to an exists folder. This still needs to be documented on the cgi-node site: update CgiNodeConfig.SessionPath path within cgi-node.js or cgi-node.min.js to point to an existing folder.
- If you download version v0.2 of cgi-node.js then use CgiNodeInfo() instead of cgiNodeInfo().

I hope this helps and let me know if the issue persists.

**Comment by Placeofcase** [comment]
Thank you UeiRicho that cleared it up for me!

**Comment by arttay** [comment]
Just leaving a thank you note. This info helped me quite a bit.

**Comment by orendayan** [comment]
Thanks, this info helped a lot

**Comment by gooeywizard** [comment]
Thanks UeiRicho, your last comment fixed all the issues I was having. It would be really helpful if the [tutorial](http://www.cgi-node.org/docs/cgi-node-on-godaddy-tutorial#TOC-Download-Node.js) could be updated with this new information.

**Comment by MahendraTanveer** [comment]
Thanks a lot UeiRicho, your solution saved my lot of time. Trying node.js for very first time. Cheers.

**Comment by lthibeault** [comment]
I am having this issue still, I am running 
Shared Hosting, with  multiple domains and subdomains in CPanel on godaddy.com
I recieve a 400 and 404 error when i try running the info script.
I have read the information above, and verified the "Account Name Line"
I have also created a generic PHP page so as to verify connectivity with the subdomain i have created for testing this. Any assistance would be wonderful.
thanks in advance

**Comment by StaticSpark** [comment]
So when I called GoDaddy, (who aquired nodejitsu). They told me because my env was in Shared Hosting I could not use node either. I am also seeing similar answers on stackoverflow. http://stackoverflow.com/questions/17529613/why-node-js-cant-run-on-shared-hosting
I think my next step will be looking into nodephp.

**Comment by mleden** [comment]
Thank you for the details here.  Similar to others on this thread, including OP, I was trying to follow "Run CGI-Node.js on GoDaddy Tutorial".  This thread helped me work around a few ambiguities/issues with those instructions.

**Comment by gooeywizard** [comment]
I tried getting this to work, but ended up giving up. Instead, I set up a free account on Openshift and installed nodejs there. So far I am super happy with it (though getting the domains on godaddy to point to my openshift app took some doing, because godaddy makes it difficult).

**Comment by jondarrer** [comment]
Thanks @UeiRicho your tips have done the trick!

**Comment by lustrabit** [comment]
Hi all,

its suppose that everything is fine, but i have a question regarding, if this is nodejs code, i tried some examples but  are not working as my local machine.
Could you please help me regarding that step?

**Comment by amit1235813** [comment]
It is still showing a 500 error for me.

I have used the Linux 64bit version of node.js.
Here is the path on cgi-node.js alongwith the path to the nodeSessions folder:
![screen shot 2017-01-05 at 2 02 47 pm](https://cloud.githubusercontent.com/assets/8248501/21673724/4246d64e-d350-11e6-9d74-3491a5d61c1c.png)

I have downloaded version v0.2 of cgi-node.js and using CgiNodeInfo() instead of cgiNodeInfo().

The error still persists:
![screen shot 2017-01-05 at 2 09 52 pm](https://cloud.githubusercontent.com/assets/8248501/21673817/cb544f5c-d350-11e6-8f75-fc43fbda60aa.png)

Any help?
